app/models/Register.scala

```diff
@@ -0,0 +1,228 @@
+    Some(AddressNoId(
+      addressLine1 = address.addressLine1,
+      addressLine2 = Some(address.addressLine2),
+      addressLine3 = address.addressLine3.get,
+      addressLine4 = address.addressLine4,
+      postalCode = address.postCode,
+      countryCode = address.country.code
+    ))
+
+
+  implicit lazy val writes: OWrites[AddressNoId] = OWrites[AddressNoId] {
+    address =>
+      Json.obj(
+        "addressLine1" -> address.addressLine1,
+        "addressLine2" -> address.addressLine2,
+        "addressLine3" -> address.addressLine3,
+        "addressLine4" -> address.addressLine4,
+        "postalCode" -> address.postalCode,
+        "countryCode" -> address.countryCode
+      )
+  }
+
+  implicit lazy val reads: Reads[AddressNoId] = {
+    import play.api.libs.functional.syntax._
+    (
+      (__ \ "addressLine1").read[String] and
+        (__ \ "addressLine2").readNullable[String] and
+        (__ \ "addressLine3").read[String] and
+        (__ \ "addressLine4").readNullable[String] and
+        (__ \ "postalCode").readNullable[String] and
+        (__ \ "countryCode").read[String]
+      )((a1, a2,a3, a4, pc, cc) => AddressNoId(a1, a2,a3, a4, pc, cc))
+  }
+}
+
+case class ContactDetails(phoneNumber: Option[String], mobileNumber: Option[String], faxNumber: Option[String], emailAddress: Option[String])
+
+object ContactDetails {
+  implicit lazy val writes: OWrites[ContactDetails] = OWrites[ContactDetails] {
+    details =>
+      Json.obj(
+        "phoneNumber" -> details.phoneNumber,
+        "mobileNumber" -> details.mobileNumber,
+        "faxNumber" -> details.faxNumber,
+        "emailAddress" -> details.emailAddress
+      )
+  }
+
+  implicit lazy val reads: Reads[ContactDetails] = {
+    import play.api.libs.functional.syntax._
+    (
+      (__ \ "phoneNumber").readNullable[String] and
+        (__ \ "mobileNumber").readNullable[String] and
+        (__ \ "faxNumber").readNullable[String] and
+        (__ \ "emailAddress").readNullable[String]
+      )((phone, mobile,fax, email) => ContactDetails(phone, mobile,fax, email))
+  }
+}
+
+case class Identification(idNumber: String, issuingInstitution: String, issuingCountryCode: String)
+
+object Identification {
+  implicit lazy val writes: OWrites[Identification] = OWrites[Identification] {
+    id =>
+      Json.obj(
+        "idNumber" -> id.idNumber,
+        "issuingInstitution" -> id.issuingInstitution,
+        "issuingCountryCode" -> id.issuingCountryCode
+      )
+  }
+
+  implicit lazy val reads: Reads[Identification] = {
+    import play.api.libs.functional.syntax._
+    (
+      (__ \ "idNumber").read[String] and
+        (__ \ "issuingInstitution").read[String] and
+        (__ \ "issuingCountryCode").read[String]
+      )((number, institution, cc) => Identification(number, institution, cc))
+  }
+}
+
+case class RequestCommon(receiptDate: String, regime:String, acknowledgementRef: String, parameters: Option[String])
```

Comment:
This isn't right, parameters is an Option of Seq[RequestParameters] you can look at my pr for an example